##[error]Error: The process 'sudo' failed with exit code 1

I am getting this error in your code.

```
sudo bundle install --deployment
/usr/lib/ruby/2.5.0/rubygems.rb:289:in `find_spec_for_exe': can't find gem bundler (>= 0.a) with executable bundle (Gem::GemNotFoundException)
	from /usr/lib/ruby/2.5.0/rubygems.rb:308:in `activate_bin_path'
	from /usr/local/bin/bundle:23:in `<main>'
##[error]Error: The process 'sudo' failed with exit code 1
##[error]Node run failed with exit code 1
```

--- a/README.md
+++ b/README.md
@@ -34,20 +34,32 @@ $ openssl enc -aes-256-cbc -md sha512 -salt -in .ssh/id_rsa -out deploy_id_rsa_e
 
 ## Workflow example
 ```yaml
-name: deploy_production
+name: Deploy with Capistrano
 
 on:
   push:
-    branches: 
-      - master
+    branches:
+    - master
 
 jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v1
-    - uses: miloserdow/capistrano-deploy@v1.3
+    - name: Set up Ruby
+      uses: actions/setup-ruby@v1
       with:
-        target: 'production'
+        ruby-version: 2.6.5
+    - name: Restore Bundler cache
+      id: cache
+      uses: actions/cache@v1
+      with:
+        path: vendor/bundle
+        key: ${{ runner.os }}-bundle-${{ hashFiles('**/Gemfile.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-bundle-
+    - uses: miloserdow/capistrano-deploy@v2
+      with:
+        target: production
         deploy_key: ${{ secrets.DEPLOY_ENC_KEY }}
 ```

--- a/lib/run.js
+++ b/lib/run.js
@@ -25,7 +25,7 @@ function install_deps() {
     yield runner.exec();
 
     let runner2 = null;
-    runner2 = new toolrunner.ToolRunner('bundle', ['install', '--deployment']);
+    runner2 = new toolrunner.ToolRunner('bundle', ['install', '--deployment', '--jobs=4', '--without=production test']);
     yield runner2.exec();
   });
 }

--- a/lib/run.js
+++ b/lib/run.js
@@ -2,12 +2,12 @@
 
 // took some code from https://github.com/Azure/k8s-actions
 var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
-    return new (P || (P = Promise))(function (resolve, reject) {
-        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
-        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
-        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
-        step((generator = generator.apply(thisArg, _arguments || [])).next());
-    });
+  return new (P || (P = Promise))(function (resolve, reject) {
+    function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+    function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+    function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
+    step((generator = generator.apply(thisArg, _arguments || [])).next());
+  });
 };
 
 
@@ -47,16 +47,16 @@ function decrypt_key(deploy_key, enc_rsa_key_pth) {
 }
 
 function deploy(target) {
-    return __awaiter(this, void 0, void 0, function* () {
-        let args = [];
-        if (!target) {
-          args = ['exec', 'cap', 'deploy'];
-        } else {
-          args = ['exec', 'cap', target, 'deploy'];
-        }
-        let runner = new toolrunner.ToolRunner('bundle', args);
-        yield runner.exec();
-    });
+  return __awaiter(this, void 0, void 0, function* () {
+    let args = [];
+    if (!target) {
+      args = ['exec', 'cap', 'deploy'];
+    } else {
+      args = ['exec', 'cap', target, 'deploy'];
+    }
+    let runner = new toolrunner.ToolRunner('bundle', args);
+    yield runner.exec();
+  });
 }
 
 function run() {

--- a/lib/run.js
+++ b/lib/run.js
@@ -14,12 +14,9 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
 // no idea what this does
 Object.defineProperty(exports, "__esModule", { value: true });
 
-const toolCache = require("@actions/tool-cache");
 const core = require("@actions/core");
 const io = require("@actions/io");
 const toolrunner = require("@actions/exec/lib/toolrunner");
-const path = require("path");
-const fs = require("fs");
 
 function install_deps() {
     return __awaiter(this, void 0, void 0, function* () {

--- a/lib/run.js
+++ b/lib/run.js
@@ -38,11 +38,21 @@ function decrypt_key(deploy_key, enc_rsa_key_pth) {
     let runner0 = new toolrunner.ToolRunner('openssl', ['version']);
     yield runner0.exec();
 
-    // TODO: also check that the key is valid after decryption
     let runner = new toolrunner.ToolRunner('openssl',
         ['enc', '-d', '-aes-256-cbc', '-md', 'sha512', '-salt', '-in',
          enc_rsa_key_pth, '-out', 'config/deploy_id_rsa', '-k', deploy_key, '-a']);
     yield runner.exec();
+
+    let runner1 = new toolrunner.ToolRunner('chmod', ['0600', 'config/deploy_id_rsa']);
+    yield runner1.exec();
+
+    const authSock = '/tmp/ssh-auth.sock'
+    let runner2 = new toolrunner.ToolRunner('ssh-agent', ['-a', authSock]);
+    yield runner2.exec();
+
+    core.exportVariable('SSH_AUTH_SOCK', authSock);
+    let runner3 = new toolrunner.ToolRunner('ssh-add', ['config/deploy_id_rsa']);
+    yield runner3.exec();
   });
 }
 

--- a/lib/run.js
+++ b/lib/run.js
@@ -19,45 +19,15 @@ const io = require("@actions/io");
 const toolrunner = require("@actions/exec/lib/toolrunner");
 
 function install_deps() {
-    return __awaiter(this, void 0, void 0, function* () {
-        // Install Xcode for Mac OS
-        if (process.platform == 'darwin') {
-            let runner_ = new toolrunner.ToolRunner('xcode-select',  ['--install']);
-            runner_.exec();
-        } else {
-            let runner0 = new toolrunner.ToolRunner('sudo',
-                ['apt-get', 'update']);
-            runner0.exec();
-            let runner_ = new toolrunner.ToolRunner('sudo',
-                ['apt-get', 'install', 'ruby-full', 'build-essential', 'postgresql', 'libpq-dev']);
-           yield runner_.exec();
-        }
+  return __awaiter(this, void 0, void 0, function* () {
+    let runner = null;
+    runner = new toolrunner.ToolRunner('gem', ['install', 'bundler:1.17.3', '--no-document']);
+    yield runner.exec();
 
-        // WORKAROUND, TODO: Parse version from Gemfile
-        let runner = null;
-        if (process.platform == 'darwin') {
-            runner = new toolrunner.ToolRunner('gem',
-                ['install', 'bundler:1.17.2']);
-        } else {
-            runner = new toolrunner.ToolRunner('sudo',
-                ['gem', 'install', 'bundler:1.17.2']);
-        }
-        yield runner.exec();
-
-        //let runner0 = new toolrunner.ToolRunner('gem', ['install', 'capistrano', 'capistrano-rails']);
-        //yield runner0.exec();
-
-        //let runner1 = new toolrunner.ToolRunner('bundle', ['update', '--bundler']);
-        //yield runner1.exec();
-
-        let runner2 = null;
-        if (process.platform == 'darwin') {
-            runner2 = new toolrunner.ToolRunner('bundle', ['install', '--deployment']);
-        } else {
-            runner2 = new toolrunner.ToolRunner('sudo', ['bundle', 'install', '--deployment']);
-        }
-        yield runner2.exec();
-    });
+    let runner2 = null;
+    runner2 = new toolrunner.ToolRunner('bundle', ['install', '--deployment']);
+    yield runner2.exec();
+  });
 }
 
 function decrypt_key(deploy_key, enc_rsa_key_pth) {

--- a/lib/run.js
+++ b/lib/run.js
@@ -28,33 +28,33 @@ function install_deps() {
             let runner_ = new toolrunner.ToolRunner('xcode-select',  ['--install']);
             runner_.exec();
         } else {
-            let runner0 = new toolrunner.ToolRunner('sudo',  
+            let runner0 = new toolrunner.ToolRunner('sudo',
                 ['apt-get', 'update']);
             runner0.exec();
-            let runner_ = new toolrunner.ToolRunner('sudo',  
+            let runner_ = new toolrunner.ToolRunner('sudo',
                 ['apt-get', 'install', 'ruby-full', 'build-essential', 'postgresql', 'libpq-dev']);
            yield runner_.exec();
-        }  
-        
+        }
+
         // WORKAROUND, TODO: Parse version from Gemfile
         let runner = null;
         if (process.platform == 'darwin') {
-            runner = new toolrunner.ToolRunner('gem', 
+            runner = new toolrunner.ToolRunner('gem',
                 ['install', 'bundler:1.17.2']);
         } else {
-            runner = new toolrunner.ToolRunner('sudo', 
+            runner = new toolrunner.ToolRunner('sudo',
                 ['gem', 'install', 'bundler:1.17.2']);
         }
         yield runner.exec();
-        
+
         //let runner0 = new toolrunner.ToolRunner('gem', ['install', 'capistrano', 'capistrano-rails']);
         //yield runner0.exec();
-        
+
         //let runner1 = new toolrunner.ToolRunner('bundle', ['update', '--bundler']);
         //yield runner1.exec();
-        
+
         let runner2 = null;
-        if (process.platform == 'darwin') { 
+        if (process.platform == 'darwin') {
             runner2 = new toolrunner.ToolRunner('bundle', ['install', '--deployment']);
         } else {
             runner2 = new toolrunner.ToolRunner('sudo', ['bundle', 'install', '--deployment']);
@@ -67,13 +67,13 @@ function decrypt_key(deploy_key, enc_rsa_key_pth) {
   return __awaiter(this, void 0, void 0, function* () {
     // Create directory if not exists
     yield io.mkdirP('config');
-    
+
     let runner0 = new toolrunner.ToolRunner('openssl', ['version']);
     yield runner0.exec();
 
     // TODO: also check that the key is valid after decryption
-    let runner = new toolrunner.ToolRunner('openssl', 
-        ['enc', '-d', '-aes-256-cbc', '-md', 'sha512', '-salt', '-in', 
+    let runner = new toolrunner.ToolRunner('openssl',
+        ['enc', '-d', '-aes-256-cbc', '-md', 'sha512', '-salt', '-in',
          enc_rsa_key_pth, '-out', 'config/deploy_id_rsa', '-k', deploy_key, '-a']);
     yield runner.exec();
   });
@@ -97,16 +97,16 @@ function run() {
     let target = core.getInput('target');
     let deploy_key = core.getInput('deploy_key');
     let enc_rsa_key_pth = core.getInput('enc_rsa_key_pth');
-    
+
     if (!deploy_key) {
       core.setFailed('No deploy key given');
     }
-    
-    // TODO: also check that the file exists 
+
+    // TODO: also check that the file exists
     if (!enc_rsa_key_pth) {
       core.setFailed('Encrypted RSA private key undefined');
     }
-    
+
     yield install_deps();
     yield decrypt_key(deploy_key, enc_rsa_key_pth);
     yield deploy(target);

